groups_admin.rb

```diff
@@ -100,34 +125,180 @@ def update_accept_header(extension,mime_type)
+  end
+
+end
+
+post '/groups/:gid' do |gid|
+  halt 501
 end
 
 # get specific group information
 get '/groups/:gid' do |gid|
-  "get one group settings: [#{gid}]"
+
+  config = {
+      :args => gid,
+      :method_symbol => :get_group_info,
+      :handle_result => Proc.new { |result|
+        halt 404, "group not found" unless result
+        result.to_json
+      },
+      :service_name => 'ADMIN_DIRECTORY'
+  }
+
+  begin
+    run_request(config, gid)
+  rescue GGBServiceAccountError => ggb_err
+    halt ggb_err.status_code, ggb_err.cause.to_json
+  end
+
 end
 
 # delete a group
-## deal with members? / do not return existing information
 delete '/groups/:gid' do |gid|
-  "delete group: #{gid}"
+  config = {
+      :args => gid,
+      :method_symbol => :delete_group,
+      :handle_result => Proc.new { |result| result },
+      :service_name => 'ADMIN_DIRECTORY'
+  }
+
+  begin
+    run_request(config, gid)
+  rescue GGBServiceAccountError => ggb_err
+    halt ggb_err.status_code, ggb_err.cause.to_json
+  end
+
 end
 
 ############## group members
 # groups/<group id>/members (PUT POST GET DELETE)
+
+get '/groups/:gid/members' do |gid|
+  config = {
+      :args => gid,
+      :method_symbol => :list_members,
+      :handle_result => Proc.new { |result| result },
+      :service_name => 'ADMIN_DIRECTORY'
+  }
+  begin
+    run_request(config, gid)
+  rescue GGBServiceAccountError => ggb_err
+    halt ggb_err.status_code, ggb_err.cause.to_json
+  end
+end
+
+get '/groups/:gid/members/:uid' do |gid, uid|
+  config = {
+      :args => [gid, uid],
+      :method_symbol => :get_member,
+      :handle_result => Proc.new { |result| result },
+      :service_name => 'ADMIN_DIRECTORY'
+  }
+  begin
+    run_request(config, [gid, uid])
+  rescue GGBServiceAccountError => ggb_err
+    halt ggb_err.status_code, ggb_err.cause.to_json
+  end
+end
+
+put '/groups/:gid/members/:uid' do |gid, uid|
+  role = params[:role] ? params[:role] : 'MEMBER'
+  halt 400, "invalid member role: [#{role}]" unless ['MEMBER', 'OWNER'].include? role
+
+  config = {
+      :args => [gid, uid],
+      :method_symbol => :insert_member,
+      :handle_result => Proc.new { |result| result },
+      :service_name => 'ADMIN_DIRECTORY'
+  }
+
+  use_args = [gid, {'email': uid, 'role': role}]
+
+  begin
+    run_request(config, use_args)
+  rescue GGBServiceAccountError => ggb_err
+    halt ggb_err.status_code, ggb_err.cause.to_json
+  end
+end
+
+#def delete_member group_key, member_key
+delete '/groups/:gid/members/:uid' do |gid, uid|
+  config = {
+      :args => [gid, uid],
+      :method_symbol => :delete_member,
+      :handle_result => Proc.new { |result| result },
+      :service_name => 'ADMIN_DIRECTORY'
+  }
+
+  use_args = [gid, uid]
+
+  begin
+    run_request(config, use_args)
+  rescue GGBServiceAccountError => ggb_err
+    halt ggb_err.status_code, ggb_err.cause.to_json
+  end
+end
+
+post '/groups/:gid/members' do |gid|
+  halt 501, "not sensible"
```

Comment:
POST implies you don't know what to call the resource and will let the system name it.  Adding members whose names aren't well defined doesn't seem sensible.